### PR TITLE
Upgrade to Ruby 2.7

### PR DESCRIPTION
**Pull request by tpendragon**
Refs #4844

### CODE REVIEW

**Comment by tpendragon** [comment]
Confirmed this seems to work in staging.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@ def after_sign_in_path_for(resource_or_scope)
   end
 
   def guest_uid_authentication_key(key)
-    key &&= nil unless key.to_s =~ /^guest/
+    key &&= nil unless /^guest/.match?(key.to_s)
```
Comment (hackartisan) [inline app/controllers/application_controller.rb:26]: I wonder why this is preferred

Comment (hackartisan) [reply app/controllers/application_controller.rb:26]: answer: performance, apparently. https://www.rubydoc.info/gems/rubocop/0.47.1/RuboCop/Cop/Performance/RegexpMatch

File: app/services/bulk_ingest_service.rb [hunk 2]
```diff
@@ -221,6 +220,17 @@ def to_a
         nodes
       end
 
+      def mime_type(file_path)
+        basename = File.basename(file_path)
```
Comment (hackartisan) [inline app/services/bulk_ingest_service.rb:224]: do we need this line? It looks like we do this right before calling the method, and don't use file_path anywhere else in this method. just accept basename?

File: config/initializers/figgy.rb [hunk 3]
```diff
@@ -30,15 +30,13 @@ def campus_ip_ranges
...
-          IPAddr.new(address.to_s)
-        end
-      rescue
-        Rails.logger.debug("Unable to resolve #{fqdn}")
-        nil
+      address = Dnsruby::Resolver.new.query(fqdn).answer[0]&.address
+      if address.present?
+        IPAddr.new(address.to_s)
       end
+                            rescue
```
Comment (hackartisan) [inline config/initializers/figgy.rb:37]: is this really indented like this?

**Review by hackartisan — CHANGES_REQUESTED** [review]
This looks good. The style changes seem good. I had a couple of very small inline.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**